### PR TITLE
Fix source columns building in BWC branch

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Follow up to https://github.com/crate/crate/commit/f0c4e947396abf2d27b7b2c612733149ff509c3b

^ worked on master since it's already 5.4 and it doesn't have copy_to. 

In 5.3 backport PR this bug was hit (and can happen in rolling upgrades). Not backporting since already resolved this in https://github.com/crate/crate/pull/14136

### CODE REVIEW

**Review by mfussenegger — COMMENTED** [review]
Could we add a test case too?

**Comment by BaurzhanSakhariev** [comment]
> Could we add a test case too?

On master this passes anyway and for rolling upgrade we have https://github.com/crate/crate-qa/commit/fcfa4d8c75edadb050908c419e7af1a9e47af1d7 for it - it would have failed this night but tests on 5.3 caught it earlier.

**Comment by BaurzhanSakhariev** [comment]
> Could we add a test case too?

I thought https://github.com/crate/crate-qa/commit/fcfa4d8c75edadb050908c419e7af1a9e47af1d7 covers this but crate-qa didn't fail tonight 🤔 

Pushed a test

File: server/src/test/java/io/crate/metadata/doc/DocIndexMetadataTest.java [hunk 1]
```diff
@@ -1686,4 +1686,30 @@ public void test_column_of_index_has_nullable_from_real_column() throws Exceptio
...
+            .endObject()
+            .endObject();
+
+        IndexMetadata metadata = getIndexMetadata("test1", builder);
+        DocIndexMetadata md = newMeta(metadata, "test1");
+
+        assertThat(md.indices().size()).isEqualTo(1);
+        IndexReference ref = md.indices().values().iterator().next();
+        assertThat(ref.columns().size()).isEqualTo(1);
+        assertThat(ref.columns().get(0).column().fqn()).isEqualTo("description");
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/metadata/doc/DocIndexMetadataTest.java:1713]:
```suggestion
        assertThat(ref.columns()).satisfiesExactly(
            x -> assertThat(x).isReference("description")
        );
```

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**